### PR TITLE
Tools: add more categories and reorganize a bit

### PR DESCRIPTION
**Pull request by amilcarlucas**
Should be NFC, it does not add nor remove options

### CODE REVIEW

File: Tools/scripts/build_options.py [hunk 1]
```diff
@@ -62,7 +64,9 @@ def __init__(self, category, label, define, description, default, dependency):
     Feature('OSD', 'OSD_PARAM', 'OSD_PARAM_ENABLED', 'Enable OSD param', 0, 'OSD'),
     Feature('OSD', 'OSD_SIDEBARS', 'HAL_OSD_SIDEBAR_ENABLE', 'Enable Scrolling Sidebars', 0, 'OSD'),
 
-    Feature('CAN', 'PICCOLOCAN', 'HAL_PICCOLO_CAN_ENABLE', 'Enable PiccoloCAN', 0, None),
+    Feature('ESC', 'PICCOLOCAN', 'HAL_PICCOLO_CAN_ENABLE', 'Enable PiccoloCAN', 0, None),
+    Feature('ESC', 'TORQEEDO', 'HAL_TORQEEDO_ENABLED', 'Enable Torqeedo Motors', 0, None),
+
```
Comment (magicrub) [inline Tools/scripts/build_options.py:69]: What about KDECAN?

Comment (amilcarlucas) [reply Tools/scripts/build_options.py:69]: This is a re-ordering PR. Once these are in, we can start adding new ones in another PR.
And yes, KDECAN was already on my radar. I would put it under ESC and not under CAN category.

**Comment by amilcarlucas** [comment]
BTW CI tests pass.

**Comment by amilcarlucas** [comment]
@peterbarker this has no merge conflicts (yet). Can it be merged before the other stuff you are doing?

**Review by tridge — APPROVED** [review]

**Pull request merged**